### ISSUE TITLE
standalone_2/dwarf_stack_trace_dedup_test broken by 'Revert "[vm/aot/tfa] Tree shake write-only fields"'

### ISSUE THREAD

**Issue by sjindel-google** [dart-lang/sdk#40635]
See https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket.appspot.com/8888516573117058848/+/steps/test_results/0/logs/new_test_failures__logs_/0

**Comment by alexmarkov** [comment]
@sstrickl do you know why `tests/standalone_2/dwarf_stack_trace_dedup_test` depends on the removal of fields by the tree shaker and fails specifically in bare instructions mode? Is it exposing a bug in dwarf stack traces, or the test is not reliable wrt changes in the tree shaker?

**Comment by sstrickl** [comment]
Well, with `--dedup-instructions`, if the instructions corresponding to the test functions main, foo, or bar are deduplicated with another Code object... that's the only cases where this test will end up failing. Looking at the stack trace info decoded, then it looks like `_stderrFD` and `foo` may have ended up with the same instructions in the deduplicator.

I'm not sure how well, if at all, DWARF works with the same virtual address being mapped to different names. If that can be encoded fine, then our DWARF reader needs to handle the possibility of the same address being mapped to multiple names. I'll take a look and see what I can find.

Oh, and it's specifically failing in _non-bare_ mode. That means we're running `DedupInstructions`, not `DedupInstructionsWithSameMetadata`. The latter requires the `Code` objects containing the `Instructions` to match certain criteria before the `Instructions` can be deduplicated, whereas the former just deduplicates as long as the instructions payloads are bit-wise equal. Maybe the assumption we can do that without checking the metadata is faulty in some way? /cc @mkustermann )

**Comment by mkustermann** [comment]
Instruction deduplication (plus dwarf stacks, bare mode or both) is can cause frames to be misattributed, yes.

**Comment by sstrickl** [comment]
Right, so we can never really know whether this test will succeed or not when instructions are being deduplicated.

Thus, I've created [CL 136122](https://dart-review.googlesource.com/c/sdk/+/136122), which removes this test and instead adds a second run of `corelib/apply2_test` that uses the `--dedup-instructions` flag. I chose that specific test because it was the one I used to debug the issues in instruction deduplication when paired with RawInstruction object removal.
